test/integration/server-selection/server_selection_operation_count.prose.test.ts

```diff
@@ -0,0 +1,157 @@
+import { expect } from 'chai';
+import { promisify } from 'util';
+
+import { CommandStartedEvent } from '../../../src';
+import { Collection } from '../../../src/collection';
+import { MongoClient } from '../../../src/mongo_client';
+
+const failPoint = {
+  configureFailPoint: 'failCommand',
+  mode: { times: 10000 },
+  data: {
+    failCommands: ['find'],
+    blockConnection: true,
+    blockTimeMS: 500,
+    appName: 'loadBalancingTest'
+  }
+};
+
+const POOL_SIZE = 100;
+
+async function runTaskGroup(collection: Collection, count: 10 | 100) {
+  for (let i = 0; i < count; ++i) {
+    await collection.findOne({});
+  }
+}
+
+async function ensurePoolIsFull(client: MongoClient) {
+  let connectionCount = 0;
+  const onConnectionCreated = () => connectionCount++;
+  client.on('connectionCreated', onConnectionCreated);
+
+  // 250ms should be plenty of time to fill the connection pool,
+  // but just in case we'll loop a couple of times.
+  for (let i = 0; connectionCount < POOL_SIZE * 2 && i < 10; ++i) {
+    await promisify(setTimeout)(250);
+  }
+
+  client.removeListener('connectionCreated', onConnectionCreated);
+
+  if (connectionCount !== POOL_SIZE * 2) {
+    throw new Error('Connection pool did not fill up');
+  }
+}
+const TEST_METADATA: MongoDBMetadataUI = { requires: { mongodb: '>=4.2.9', topology: 'sharded' } };
+
+describe('Server Selection Operation Count Prose', function () {
```

Comment:
I'll update the test name (topical comment, given our discussion of testing conventions this morning)! 

As far as the second part goes - is the ask to rename the file `server_selection.prose.operation_count.test.ts`?  The test is already broken out into its own prose test file.

Reply:
Sorry for the confusion, I meant, is it even worth having it in its own file right now vs in a general `server_selection.prose.test.ts` - basically, I was saying that it might be fine to just keep it as `server_selection.prose.test.ts` for now and not try to give it its own special name. But, I'm not strongly opposed to leaving it in a dedicated file as long as we follow the .prose.<specific test>.test convention.